src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Functions/Processor.cs

```diff
@@ -58,52 +62,43 @@ IDataArchiveService dataArchiveService
             [Queue("imports-available")] ICollector<ImportMessage> collector
         )
         {
-            if (message.Seeding)
-            {
-                await _batchService.CreateImport(message.Release.Id.ToString(), message.DataFileName, 0, message);
-            }
-
             if (message.ArchiveFileName != "")
             {
-                await _batchService.UpdateStatus(message.Release.Id.ToString(), message.DataFileName, IStatus.PROCESSING_ARCHIVE_FILE);
+                await _importStatusService.UpdateStatus(message.Release.Id, message.DataFileName, IStatus.PROCESSING_ARCHIVE_FILE);
                 await _dataArchiveService.ExtractDataFiles(message.Release.Id, message.ArchiveFileName);
             }
 
-            await _batchService.UpdateStatus(message.Release.Id.ToString(), message.DataFileName, IStatus.RUNNING_PHASE_1);
+            await _importStatusService.UpdateStatus(message.Release.Id, message.DataFileName, IStatus.STAGE_1);
             var subjectData = await _fileStorageService.GetSubjectData(message);
 
             await _validatorService.Validate(message.Release.Id, subjectData, executionContext, message)
                 .OnSuccess(async result =>
                 {
                     try
                     {
+                        var status = await _importStatusService.GetImportStatus(message.Release.Id, message.OrigDataFileName);
                         message.RowsPerBatch = result.RowsPerBatch;
                         message.TotalRows = result.FilteredObservationCount;
                         message.NumBatches = result.NumBatches;
 
                         await _batchService.UpdateStoredMessage(message);
-                        
-                        var status = await _batchService.GetStatus(message.Release.Id.ToString(), message.OrigDataFileName);
 
-                        // If already reached Phase 2 then don't re-create the subject or split into batches
-                        if ((int) status < (int) IStatus.RUNNING_PHASE_2)
+                        // If already reached Phase 4 then don't re-create the subject or split into batches
+                        if ((int) status.Status < (int) IStatus.STAGE_4)
```

Comment:
Comparing the enums like this feels a bit icky. Perhaps `CompareTo` would be nicer?

Reply:
I've used CompareTo now but it's slightly complicated now by needing to check for Stage 3 being incomplete.